src/azure-cli-core/azure/cli/core/tests/test_profile.py

```diff
@@ -645,7 +645,7 @@ def test_get_login_credentials_aux_tenants(self, mock_get_token, mock_read_cred_
                                                                      aux_tenants=[test_tenant_id2])
 
     @mock.patch('azure.cli.core._profile._load_tokens_from_file', autospec=True)
-    @mock.patch('msrestazure.azure_active_directory.MSIAuthentication', autospec=True)
+    @mock.patch('azure.cli.core._profile.MSIAuthenticationWrapper', autospec=True)
```

Comment:
Is this change necessary? Won't it be nice to have `MSIAuthenticationWrapper` also tested?

Reply:
`mock.patch` by default will create a `MagicMock` instance which can't be inherited, maybe using `new=MSIAuthentication` may work? Haven't tested.